server/leaderboard.js

```diff
@@ -27,23 +27,125 @@ var users = require('./users.js');
  *
  * If shrt is true, return leaderboard with max eight entries.
  */
-exports.leaderboard = function(userid, shrt, callback) {
+exports.leaderboard = function(userid, shrt, type, callback) {
     users.getStudentsListSorted(0, function(err, studentlist) {
+
         var rank = 0;
         var last = -1;
         var userind;
 
+        //Sort the list of students depending on the type (defualt sorting is by points)
+        if (type === 'accuracy'){
+            studentlist = studentlist.sort(function(a,b) {
+                x = a.correctAttemptsCount/a.totalAttemptsCount;
+                y = b.correctAttemptsCount/b.totalAttemptsCount;
+
+                if (a.totalAttemptsCount === 0 && b.totalAttemptsCount ===0){
```

Comment:
seems like there is a good amount of duplicated code here, this should be moved in a helper function

Reply:
Good point